ion_hash_test.go

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ionhash
+
+import (
+	"bytes"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/amzn/ion-go/ion"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIonHash(t *testing.T) {
+	parameters := ionHashDataSource(t)
+
+	for i := range parameters {
+		ionBinary := parameters[i].testCase
+		reader := ion.NewReaderBytes(ionBinary)
+
+		provider := parameters[i].provider
+		Traverse(t, reader, provider.getInstance())
+
+		if len(parameters[i].expectedHashLog.identityUpdateList) > 0 {
+			assert.Equal(t, parameters[i].expectedHashLog.identityUpdateList, provider.getUpdateHashLog(), parameters[i].hasherName+" failed")
+		}
+		if len(parameters[i].expectedHashLog.identityDigestList) > 0 {
+			assert.Equal(t, parameters[i].expectedHashLog.identityDigestList, provider.getDigestHashLog(), parameters[i].hasherName+" failed")
+		}
+		if len(parameters[i].expectedHashLog.identityFinalDigestList) > 0 {
+			assert.Equal(t, parameters[i].expectedHashLog.identityFinalDigestList, provider.getDigestHashLog(), parameters[i].hasherName+" failed")
+		}
+
+		if len(parameters[i].expectedHashLog.md5UpdateList) > 0 {
+			assert.Equal(t, parameters[i].expectedHashLog.md5UpdateList, provider.getDigestHashLog(), parameters[i].hasherName+" failed")
+		}
+		if len(parameters[i].expectedHashLog.md5DigestList) > 0 {
+			assert.Equal(t, parameters[i].expectedHashLog.md5DigestList, provider.getDigestHashLog(), parameters[i].hasherName+" failed")
+		}
+	}
+}
+
+func Traverse(t *testing.T, reader ion.Reader, provider IonHasherProvider) {
+	hr, err := NewHashReader(reader, provider)
+	require.NoError(t, err, "Something went wrong executing NewHashReader()")
+	TraverseReader(t, hr)
+	_, err = hr.Sum(nil)
+	require.NoError(t, err, "Something went wrong with executing hr.Sum()")
+}
```

Comment:
Style suggestion: I think an empty line before and after TraverseReader(t, hr) would make this a bit more readable.

Reply:
Addressed.